components/base/wheeled/wheeled_base.go

```diff
@@ -124,8 +158,173 @@ func (base *wheeledBase) Spin(ctx context.Context, angleDeg, degsPerSec float64,
 	return base.runAll(ctx, -rpm, revolutions, rpm, revolutions)
 }
 
+func (base *wheeledBase) spinWithMovementSensor(angleDeg, degsPerSec float64, extra map[string]interface{}) error {
+	wheelrpm, _ := base.spinMath(angleDeg, degsPerSec)
+
+	base.activeBackgroundWorkers.Add(1)
+	utils.ManagedGo(func() {
+		// runAll calls GoFor, which blocks until the timed operation is done, or returns nil if a zero is passed in
+		// the code inside this goroutine would block the sensor for loop if taken out
+		if err := base.runAll(base.cancelCtx, -wheelrpm, 0, wheelrpm, 0); err != nil {
+			base.logger.Error(err)
+		}
+	}, base.activeBackgroundWorkers.Done)
+
+	startYaw, err := getCurrentYaw(base.cancelCtx, base.sensors.orientation, extra)
+	if err != nil {
+		return err
+	} // from 0 -> 360
+	targetYaw, dir := findSpinParams(angleDeg, degsPerSec, startYaw)
+
+	adjustYaw := rdkutils.Float64AlmostEqual(angleDeg, float64(oneTurn), 0.1)
+	turnCount := 0
+	errCounter := 0
+
+	base.activeBackgroundWorkers.Add(1)
+	utils.ManagedGo(func() {
+		ticker := time.NewTicker(yawPollTime)
+		defer ticker.Stop()
+		for {
+			// RSDK-2384 - use a new cancel ctx for sensor loops
+			// to have them stop when cotnext is done
+			select {
+			case <-base.sensors.ctx.Done():
+				base.logger.Debug("cancelled context 1")
+				ticker.Stop()
+				return
+			default:
+			}
+			select {
+			case <-base.sensors.ctx.Done():
+				base.logger.Debug("cancelled context 2")
+				ticker.Stop()
+				return
+			case <-ticker.C:
+			}
+
+			// imu readings are limited from 0 -> 360
+			currYaw, err := getCurrentYaw(base.cancelCtx, base.sensors.orientation, extra)
+			if err != nil {
+				errCounter++
+				if errCounter > 100 {
+					base.logger.Error(errors.Wrap(
+						err, "imu sensor unreachable, 100 error counts when trying to read yaw angle"))
+					return
+				}
+			}
+			errCounter = 0 // reset reading error count to zero if we are successfully reading again
+
+			// if the imu yaw reading is close to 360, we are nearing a full turn,
+			// so we adjust the current reading by 360 * the number of turns we've done
```

Comment:
[q] why don't we have to adjust the target as well?

Reply:
nice catch, thought I did that.

Reply:
bumping this up as well?